jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java

```diff
@@ -414,15 +412,12 @@ protected void writeErrorHtmlStacks(Request request, Writer writer) throws IOExc
      * @param reason The reason for the error code (may be null)
      * @param fields The header fields that will be sent with the response.
      * @return The content as a ByteBuffer, or null for no body.
+     * @deprecated Do not override. No longer invoked by Jetty.
      */
+    @Deprecated(since = "12.0.8", forRemoval = true)
     public ByteBuffer badMessageError(int status, String reason, HttpFields.Mutable fields)
```

Comment:
Is this really not called anymore by things like the HttpParser?

Reply:
No it is not.
In Jetty 11 it was called by `HttpChannel`, but in Jetty 12 we handle failures differently by always called `HttpChannel.onFailure()`.